pipeline/splicing/calcVarPriors.py

```diff
@@ -1284,110 +1766,260 @@ def getPriorProbDeNovoDonorSNS(variant, exonicPortionSize, deNovoDonorInRefAcc=F
+                        "closestRefSeq": "N/A",
+                        "closestAltMaxEntScanScore": "N/A",
+                        "closestAltZScore": "N/A",
+                        "closestAltSeq": "N/A",
+                        "closestExonStart": "N/A",
+                        "closestIntronStart": "N/A",
+                        "frameshiftFlag": "N/A",
+                        "altGreaterRefFlag": "N/A",
+                        "altGreaterClosestRefFlag": "N/A",
+                        "altGreaterClosestAltFlag": "N/A"}
+            slidingWindowInfo = getMaxMaxEntScanScoreSlidingWindowSNS(variant, exonicPortionSize, STD_DE_NOVO_LENGTH,
+                                                                      donor=True, deNovoDonorInRefAcc=deNovoDonorInRefAcc)
+            # +-1 because per HCI PRIORS website donor position is defined as being first nucleotide that is NOT included in spliced exon
+            if getVarStrand(variant) == "-":
+                 newGenomicSplicePos = getNewSplicePosition(variant["Pos"], "-", slidingWindowInfo["varWindowPosition"],
+                                                            slidingWindowInfo["inExonicPortion"], STD_EXONIC_PORTION, STD_ACC_INTRONIC_LENGTH,
+                                                            donor=True) - 1
+            else:
+                newGenomicSplicePos = getNewSplicePosition(variant["Pos"], "+", slidingWindowInfo["varWindowPosition"],
+                                                            slidingWindowInfo["inExonicPortion"], STD_EXONIC_PORTION, STD_ACC_INTRONIC_LENGTH,
+                                                            donor=True) + 1
+            deNovoOffset = 0
+            subDonorInfo = getClosestSpliceSiteScores(variant, deNovoOffset, donor=True, deNovo=False,
+                                                      deNovoDonorInRefAcc=deNovoDonorInRefAcc)
+            subZScore = subDonorInfo["zScore"]
+            refAltZScore = "N/A"
+            refAltMES = "N/A"
+            refAltSeq = "N/A"
+            altGreaterClosestRefFlag = 0
+            altGreaterClosestAltFlag = "N/A"
+            if varInSpliceRegion(variant, donor=True, deNovo=False) == True:
+                refDonorInfo = getPriorProbRefSpliceDonorSNS(variant, "enigma")
+                refAltZScore = refDonorInfo["altZScore"]
+                refAltMES = refDonorInfo["altMaxEntScanScore"]
+                refAltSeq = refDonorInfo["altSeq"]
+                altGreaterClosestAltFlag = 0
+            frameshiftFlag = 0
+            frameshiftStatus = getDeNovoSpliceFrameshiftStatus(variant, donor=True, deNovoDonorInRefAcc=deNovoDonorInRefAcc)
+            if frameshiftStatus == True:
+                frameshiftFlag = 1
+            altZScore = slidingWindowInfo["altZScore"]
+            refZScore = slidingWindowInfo["refZScore"]
+            altGreaterRefFlag = 0
             if altZScore <= refZScore:
-                priorProb = 0
+                priorProb = 0.02
```

Comment:
as before, can all of these values be stored in constants somewhere or would that cause more problems than it would solve?

Reply:
Stored now at the beginning